lib/AST/PlatformKind.cpp

```diff
@@ -24,6 +24,15 @@
 using namespace swift;
 
 StringRef swift::platformString(PlatformKind platform) {
+  // FIXME: Update PlatformKinds.def to use the macOS spelling by default.
```

Comment:
TBH, it's probably a good idea to just bite the bullet and do this, rather than continuing to add more special cases everywhere.

Reply:
Done. @brentdax do you think this change would be ok for 5.3, or too risky?

Reply:
I don’t think there’s much additional risk from implementing it this way:

* By far the riskiest thing in this PR is changing the string we print into machine-readable files, because we may not have test coverage proving that the compiler accepts “macOS” everywhere it would accept “OSX”, and we certainly can’t prove that all non-compiler tools that work with these files will accept it. But both implementations do this, so changing PlatformKinds.def has no bearing on it.

* Changing the contents of PlatformKinds.def creates additional risk in the places where we actually include PlatformKinds.def, but it’s trivial to audit these (there are only four) and see that they’re all correct.

    * We should verify that there’s test coverage of both `”OSX”` and `”macOS”` being fed into `swift::platformFromString()`, but I have little doubt that there is.

* There’s no additional risk at places where we merely use `PlatformKind::OSX` because the compiler verifies that all of those are changed to `PlatformKind::macOS`. (However, this could cause mis-merges/merge conflicts for a little while.)

Reply:
Thanks Brent! Your first point does sound a little worrying. I think I'll see if I can make the spelling change specific to sourcekitd's interface generation for 5.3 to avoid that risk and leave this more comprehensive spelling change as master-only.